### ISSUE TITLE
function.name doesn't return name if function is not defer parsed

### ISSUE THREAD

**Issue by kunalspathak** [Microsoft/ChakraCore#1128]
```js
var obj = {};
obj.foo = function () {
    print(obj.foo.name);
}
obj.foo();
```
 
Above test case prints function name only if function is defer parsed.
 
```cmd
ch.exe –Es6Experimental –forceDeferParse test.js
Prints “foo”

ch.exe –Es6Experimental test.js
Prints nothing.
 ```

**Comment by kunalspathak** [comment]
FYI - @pleath , @akroshg , @abchatra

**Comment by akroshg** [comment]
-forcedeferparse is incorrect. https://tc39.github.io/ecma262/#sec-assignment-operators-runtime-semantics-evaluation
We should assign the name when the LHS is identifier ref.

**Comment by akroshg** [comment]
Fixed by #1151
